transactor/gas_tracker.go

```diff
@@ -0,0 +1,77 @@
+package transactor
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/rs/zerolog/log"
+	"io/ioutil"
+	"net/http"
+)
+
+var ErrFailToGetResponseFromGasTracker = errors.New("failed to get a response from the gas tracker")
+
+// GasTracker provides methods for gas tracking
+type GasTracker interface {
+	// GetSuggestedGasPriceFromGasTracker retrieve the network's suggested gas price
+	GetSuggestedGasPriceFromGasTracker(ctx context.Context) (*GasTrackerResponse, error)
```

Comment:
i think we can simplify the name of the function to `GetSuggestedGasPrice`, it is self explanatory that it will get the suggested price from the tracker

Reply:
fixed with 84dc1689f2db7b708f94589aca699be6ca83a39c